// plugin-welcome/src/api/models/ControllersUser.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface ControllersUser
 */
export interface ControllersUser {
    /**
     * 
     * @type {string}
     * @memberof ControllersUser
     */
    confirmPassword?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersUser
     */
    dateOfBirth?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersUser
     */
    email?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersUser
     */
    firstName?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersUser
     */
    gender?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersUser
     */
    identityCard?: number;
    /**
     * 
     * @type {string}
     * @memberof ControllersUser
     */
    lastName?: string;
    /**
     * 
     * @type {string}
     * @memberof ControllersUser
     */
    password?: string;
    /**
     * 
     * @type {number}
     * @memberof ControllersUser
     */
    phone?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersUser
     */
    province?: number;
    /**
     * 
     * @type {number}
     * @memberof ControllersUser
     */
    userType?: number;
}

export function ControllersUserFromJSON(json: any): ControllersUser {
    return ControllersUserFromJSONTyped(json, false);
}

export function ControllersUserFromJSONTyped(json: any, ignoreDiscriminator: boolean): ControllersUser {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'confirmPassword': !exists(json, 'confirmPassword') ? undefined : json['confirmPassword'],
        'dateOfBirth': !exists(json, 'dateOfBirth') ? undefined : json['dateOfBirth'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'firstName': !exists(json, 'firstName') ? undefined : json['firstName'],
        'gender': !exists(json, 'gender') ? undefined : json['gender'],
        'identityCard': !exists(json, 'identityCard') ? undefined : json['identityCard'],
        'lastName': !exists(json, 'lastName') ? undefined : json['lastName'],
        'password': !exists(json, 'password') ? undefined : json['password'],
        'phone': !exists(json, 'phone') ? undefined : json['phone'],
        'province': !exists(json, 'province') ? undefined : json['province'],
        'userType': !exists(json, 'userType') ? undefined : json['userType'],
    };
}

export function ControllersUserToJSON(value?: ControllersUser | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'confirmPassword': value.confirmPassword,
        'dateOfBirth': value.dateOfBirth,
        'email': value.email,
        'firstName': value.firstName,
        'gender': value.gender,
        'identityCard': value.identityCard,
        'lastName': value.lastName,
        'password': value.password,
        'phone': value.phone,
        'province': value.province,
        'userType': value.userType,
    };
}
