// plugin-welcome/src/api/models/EntGender.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntGenderEdges,
    EntGenderEdgesFromJSON,
    EntGenderEdgesFromJSONTyped,
    EntGenderEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntGender
 */
export interface EntGender {
    /**
     * GenderName holds the value of the "Gender_Name" field.
     * @type {string}
     * @memberof EntGender
     */
    genderName?: string;
    /**
     * 
     * @type {EntGenderEdges}
     * @memberof EntGender
     */
    edges?: EntGenderEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntGender
     */
    id?: number;
}

export function EntGenderFromJSON(json: any): EntGender {
    return EntGenderFromJSONTyped(json, false);
}

export function EntGenderFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntGender {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'genderName': !exists(json, 'Gender_Name') ? undefined : json['Gender_Name'],
        'edges': !exists(json, 'edges') ? undefined : EntGenderEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntGenderToJSON(value?: EntGender | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'Gender_Name': value.genderName,
        'edges': EntGenderEdgesToJSON(value.edges),
        'id': value.id,
    };
}
