// plugin-welcome/src/api/models/EntProvince.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntProvinceEdges,
    EntProvinceEdgesFromJSON,
    EntProvinceEdgesFromJSONTyped,
    EntProvinceEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntProvince
 */
export interface EntProvince {
    /**
     * ProvinceName holds the value of the "Province_Name" field.
     * @type {string}
     * @memberof EntProvince
     */
    provinceName?: string;
    /**
     * 
     * @type {EntProvinceEdges}
     * @memberof EntProvince
     */
    edges?: EntProvinceEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntProvince
     */
    id?: number;
}

export function EntProvinceFromJSON(json: any): EntProvince {
    return EntProvinceFromJSONTyped(json, false);
}

export function EntProvinceFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntProvince {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'provinceName': !exists(json, 'Province_Name') ? undefined : json['Province_Name'],
        'edges': !exists(json, 'edges') ? undefined : EntProvinceEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntProvinceToJSON(value?: EntProvince | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'Province_Name': value.provinceName,
        'edges': EntProvinceEdgesToJSON(value.edges),
        'id': value.id,
    };
}
