// plugin-welcome/src/api/models/EntProvinceEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntUser,
    EntUserFromJSON,
    EntUserFromJSONTyped,
    EntUserToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntProvinceEdges
 */
export interface EntProvinceEdges {
    /**
     * ProvinceID holds the value of the Province_ID edge.
     * @type {Array<EntUser>}
     * @memberof EntProvinceEdges
     */
    provinceID?: Array<EntUser>;
}

export function EntProvinceEdgesFromJSON(json: any): EntProvinceEdges {
    return EntProvinceEdgesFromJSONTyped(json, false);
}

export function EntProvinceEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntProvinceEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'provinceID': !exists(json, 'provinceID') ? undefined : ((json['provinceID'] as Array<any>).map(EntUserFromJSON)),
    };
}

export function EntProvinceEdgesToJSON(value?: EntProvinceEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'provinceID': value.provinceID === undefined ? undefined : ((value.provinceID as Array<any>).map(EntUserToJSON)),
    };
}
