// plugin-welcome/src/api/models/EntUserTypeEdges.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntUser,
    EntUserFromJSON,
    EntUserFromJSONTyped,
    EntUserToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntUserTypeEdges
 */
export interface EntUserTypeEdges {
    /**
     * UserTypeID holds the value of the UserType_ID edge.
     * @type {Array<EntUser>}
     * @memberof EntUserTypeEdges
     */
    userTypeID?: Array<EntUser>;
}

export function EntUserTypeEdgesFromJSON(json: any): EntUserTypeEdges {
    return EntUserTypeEdgesFromJSONTyped(json, false);
}

export function EntUserTypeEdgesFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntUserTypeEdges {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'userTypeID': !exists(json, 'userTypeID') ? undefined : ((json['userTypeID'] as Array<any>).map(EntUserFromJSON)),
    };
}

export function EntUserTypeEdgesToJSON(value?: EntUserTypeEdges | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'userTypeID': value.userTypeID === undefined ? undefined : ((value.userTypeID as Array<any>).map(EntUserToJSON)),
    };
}
