// plugin-welcome/src/api/models/EntUserType.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntUserTypeEdges,
    EntUserTypeEdgesFromJSON,
    EntUserTypeEdgesFromJSONTyped,
    EntUserTypeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntUserType
 */
export interface EntUserType {
    /**
     * UserTypeName holds the value of the "UserType_name" field.
     * @type {string}
     * @memberof EntUserType
     */
    userTypeName?: string;
    /**
     * 
     * @type {EntUserTypeEdges}
     * @memberof EntUserType
     */
    edges?: EntUserTypeEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntUserType
     */
    id?: number;
}

export function EntUserTypeFromJSON(json: any): EntUserType {
    return EntUserTypeFromJSONTyped(json, false);
}

export function EntUserTypeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntUserType {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'userTypeName': !exists(json, 'UserType_name') ? undefined : json['UserType_name'],
        'edges': !exists(json, 'edges') ? undefined : EntUserTypeEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntUserTypeToJSON(value?: EntUserType | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'UserType_name': value.userTypeName,
        'edges': EntUserTypeEdgesToJSON(value.edges),
        'id': value.id,
    };
}
