// plugin-welcome/src/api/models/EntUser.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntUserEdges,
    EntUserEdgesFromJSON,
    EntUserEdgesFromJSONTyped,
    EntUserEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntUser
 */
export interface EntUser {
    /**
     * ConfirmPassword holds the value of the "Confirm_password" field.
     * @type {string}
     * @memberof EntUser
     */
    confirmPassword?: string;
    /**
     * DateOfBirth holds the value of the "Date_of_birth" field.
     * @type {number}
     * @memberof EntUser
     */
    dateOfBirth?: number;
    /**
     * Email holds the value of the "Email" field.
     * @type {string}
     * @memberof EntUser
     */
    email?: string;
    /**
     * FirstName holds the value of the "First_name" field.
     * @type {string}
     * @memberof EntUser
     */
    firstName?: string;
    /**
     * IdentityCard holds the value of the "Identity_card" field.
     * @type {number}
     * @memberof EntUser
     */
    identityCard?: number;
    /**
     * LastName holds the value of the "Last_name" field.
     * @type {string}
     * @memberof EntUser
     */
    lastName?: string;
    /**
     * Password holds the value of the "Password" field.
     * @type {string}
     * @memberof EntUser
     */
    password?: string;
    /**
     * Phone holds the value of the "Phone" field.
     * @type {number}
     * @memberof EntUser
     */
    phone?: number;
    /**
     * 
     * @type {EntUserEdges}
     * @memberof EntUser
     */
    edges?: EntUserEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntUser
     */
    id?: number;
}

export function EntUserFromJSON(json: any): EntUser {
    return EntUserFromJSONTyped(json, false);
}

export function EntUserFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntUser {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'confirmPassword': !exists(json, 'Confirm_password') ? undefined : json['Confirm_password'],
        'dateOfBirth': !exists(json, 'Date_of_birth') ? undefined : json['Date_of_birth'],
        'email': !exists(json, 'Email') ? undefined : json['Email'],
        'firstName': !exists(json, 'First_name') ? undefined : json['First_name'],
        'identityCard': !exists(json, 'Identity_card') ? undefined : json['Identity_card'],
        'lastName': !exists(json, 'Last_name') ? undefined : json['Last_name'],
        'password': !exists(json, 'Password') ? undefined : json['Password'],
        'phone': !exists(json, 'Phone') ? undefined : json['Phone'],
        'edges': !exists(json, 'edges') ? undefined : EntUserEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntUserToJSON(value?: EntUser | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'Confirm_password': value.confirmPassword,
        'Date_of_birth': value.dateOfBirth,
        'Email': value.email,
        'First_name': value.firstName,
        'Identity_card': value.identityCard,
        'Last_name': value.lastName,
        'Password': value.password,
        'Phone': value.phone,
        'edges': EntUserEdgesToJSON(value.edges),
        'id': value.id,
    };
}
